qiskit/pulse/commands/command.py

```diff
@@ -42,11 +52,33 @@ def __init__(self, duration: int = None, name: str = None):
         else:
             raise PulseError('Pulse duration should be integer.')
 
-        if name:
-            self._name = name
+        self._name = Command.create_name()
+
+    @classmethod
+    def create_name(cls, name: str = None) -> str:
+        """Method to create names for pulse commands."""
+        if name is None:
+            try:
+                name = '%s%i' % (cls.prefix, cls.instances_counter)
+            except TypeError:
+                raise PulseError("prefix and counter must be non-None when name is None.")
         else:
-            self._name = 'p%d' % Command.pulseIndex
-            Command.pulseIndex += 1
+            try:
+                name = str(name)
+            except Exception:
+                raise PulseError("The pulse command name should be castable to a string "
+                                 "(or None for autogenerate a name).")
+            name_format = re.compile('[a-z][a-zA-Z0-9_]*')
+            if name_format.match(name) is None:
+                raise PulseError("%s is an invalid OpenPulse command name." % name)
+
+        cls.increment_count()
+
+        return name
+
+    @classmethod
```

Comment:
Not blocking, but I don't really think there is a need to have this as a separate classmethod since it is only called within `create_name`.

Reply:
Good point. I'm fine with moving it to create_name. I'll make an extra commit to fix this and disable pylint.